app/Helpers/database/hash.php

```diff
@@ -1,142 +1,91 @@
 <?php
 
-use App\Community\Enums\ArticleType;
+use App\Models\Game;
+use App\Models\GameHash;
 
-function getMD5List(int $consoleID): array
+function getMD5List(int $consoleId): array
 {
-    $retVal = [];
-
-    $whereClause = "";
-    if ($consoleID > 0) {
-        $whereClause = "WHERE gd.ConsoleID = $consoleID ";
-    }
-
-    $query = "SELECT MD5, GameID
-              FROM GameHashLibrary AS ghl
-              LEFT JOIN GameData AS gd ON gd.ID = ghl.GameID
-              $whereClause
-              ORDER BY GameID ASC";
-
-    foreach (legacyDbFetchAll($query) as $nextData) {
-        $nextData['GameID'] = (int) $nextData['GameID'];
-        $retVal[$nextData['MD5']] = $nextData['GameID'];
-    }
-
-    return $retVal;
+    $query = GameHash::query()
+        ->select('game_hashes.md5', 'game_hashes.game_id')
+        ->leftJoin('GameData as gd', 'gd.ID', '=', 'game_hashes.game_id')
+        ->when($consoleId > 0, function ($q) use ($consoleId) {
+            $q->where('gd.ConsoleID', $consoleId);
+        })
+        ->orderBy('game_hashes.game_id', 'asc');
+
+    return $query->pluck('game_id', 'md5')->toArray();
 }
 
-function getHashListByGameID(int $gameID): array
+function getHashListByGameID(int $gameId): array
 {
-    if ($gameID < 1) {
+    if ($gameId < 1) {
         return [];
     }
 
-    $query = "SELECT MD5 AS Hash, Name, Labels, User
-              FROM GameHashLibrary
-              WHERE GameID = $gameID
-              ORDER BY Name, Hash";
-
-    $retVal = [];
-    $dbResult = s_mysql_query($query);
-    if ($dbResult !== false) {
-        while ($nextData = mysqli_fetch_assoc($dbResult)) {
-            $retVal[] = $nextData;
-        }
+    $game = Game::find($gameId);
+    if (!$game) {
+        return [];
     }
 
-    return $retVal;
+    $hashes = $game->hashes()
+        ->with('user')
+        ->select('md5', 'name', 'labels', 'user_id')
+        ->orderBy('name')
+        ->orderBy('md5')
+        ->get()
+        ->map(function ($hash) {
+            return [
+                'Hash' => $hash->md5,
```

Comment:
The idea was to replace the `md5` column usage with the `hash` column in `game_hashes` as `md5` might be ambiguous. Sources like `nointro` also provide md5 hashes which are not the same as the "RA" hash for game detection. `hash` would become the main reference regardless of what was hashed and how, depending on the system.

Reply:
Thanks for this additional background info. For full disclosure, this is an Eloquent ORM reimplementation of the original query:

```php
$query = "SELECT MD5 AS Hash ...
```

I figured it made sense to go ahead and migrate all of these helpers to Eloquent ORM while I was picking away in here. Agreed that migration to the `hash` field should be left to future work.

Maybe I should add a TODO to the `GameHash` model which says something like `md5` will eventually become `hash`?

Reply:
Sounds good. To be sure - the `md5` column would stay but repurposed to hold external `md5` values as soon as all contents and usage of it have been moved over to `hash`.

Reply:
How does this sound? (not committed yet)

```php
// TODO migrate functionality from md5 to hash. ensure md5 retains external md5 values and hash is the primary reference for game detection
```

Reply:
That's perfect!

Reply:
Great! Committed in latest.